carbonmark-api/src/routes/projects/get.ts

```diff
@@ -86,18 +92,26 @@ const handler = (fastify: FastifyInstance) =>
         );
         return;
       }
+      // Map the project credits to the correct map entry
       const [standard, registryProjectId] = project.projectID.split("-");
       project.carbonCredits.forEach((credit) => {
+        // Discard credits with no pool balances
+        if (credit.poolBalances.length == 0) return;
+
         const { creditId: key } = new CreditId({
           standard,
           registryProjectId,
           vintage: credit.vintage.toString(),
         });
-        ProjectMap.set(key, {
-          /** We need to remove all other credits from this asset so that it is a one to one mapping of credit to project */
-          poolProjectData: { ...project, carbonCredits: [credit] },
-          key,
-        });
+        // Create the entry if does not exist
+        if (!ProjectMap.has(key)) {
+          ProjectMap.set(key, {
+            poolProjectData: { ...project, carbonCredits: [] },
+            key,
+          });
+        }
```

Comment:
Are we hiding duplicate or multiple credits with the same key here? This goes back to the multiple credits for the same project problem from earlier.

Reply:
Hmm... Each credit should have a different key eg. `VCS-191-2009` so there would not be hidden credits if I understand this credit thing right.
There is surely a naming issue there. `projectMap` should be `creditMap` but maybe we should have a larger refactor to rename everything properly.

Reply:
Ha Ok!!!! I understand now.
Indeed, my code is hiding assets. But yeah lets do that in another PR.